torch/_refs/nn/functional/__init__.py

```diff
@@ -59,6 +60,43 @@
 
 Tensor = torch.Tensor
 
+
+@register_decomposition(torch.ops.aten.alpha_dropout)
+def alpha_dropout(
+    self: TensorLikeType, p: float = 0.5, training: bool = True, inplace: bool = False
+) -> TensorLikeType:
+
+    if inplace:
+        raise NotImplementedError
+
+    if not training:
+        return self
+
+    assert p <= 1
+    assert p >= 0
+
+    if p == 1:
+        return refs.zeros_like(self)
+
+    if p == 0:
+        return self
+
+    noise = refs.lt(
+        refs.uniform(
+            self.shape, low=0.0, high=1.0, dtype=torch.float32, device=self.device
+        ),
+        1 - p,
+    )
+
+    alpha = 1.7580993408473766
```

Comment:
Add a comment for what's happening in this section

Reply:
I added a comment.

@mruberry, please correct me if I'm wrong, but I think the alpha should be `-1.7580993408473766` instead of `1.7580993408473766` for consistency with the paper. However, the meaning is the same. Originally it was added and discussed in https://github.com/pytorch/pytorch/pull/1775#discussion_r129416665/. Later modified the alpha value in #9073.